File: backend/app/models/branch.py
from datetime import datetime, timezone

from sqlalchemy import Boolean, Column, DateTime, ForeignKey, Integer, String
from sqlalchemy.orm import relationship

from app.database import Base


class Branch(Base):
    __tablename__ = "branches"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    code = Column(String, unique=True, nullable=False, index=True)
    address = Column(String, nullable=False)
    city = Column(String, nullable=False)
    phone = Column(String, nullable=False)
    manager_id = Column(Integer, nullable=True)
    company_id = Column(Integer, ForeignKey("companies.id"), nullable=False)
    is_active = Column(Boolean, default=True, nullable=False)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), nullable=False)
    updated_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc), nullable=False)
    company = relationship("Company", back_populates="branches")
    drivers = relationship("Driver", back_populates="branch")
    company = relationship("Company", back_populates="branches")
    drivers = relationship("Driver", back_populates="branch")

File: frontend/types/branch.ts
export type Branch = {
  id: number
  name: string
  code: string
  city: string
  address: string
  phone: string
  manager_id?: number | null
  manager_name?: string | null
  is_active: boolean
  company_id: number
  created_at: string
  updated_at: string
}

export type BranchCreatePayload = {
  name: string
  code: string
  city: string
  address: string
  phone: string
  manager_name: string
  is_active: boolean
}
